pkg/machine/wsl/wutil/wutil.go

```diff
@@ -114,13 +91,50 @@ func IsWSLStoreVersionInstalled() bool {
 	return true
 }
 
-func matchOutputLine(output io.ReadCloser, match string) bool {
-	scanner := bufio.NewScanner(transform.NewReader(output, unicode.UTF16(unicode.LittleEndian, unicode.UseBOM).NewDecoder()))
-	for scanner.Scan() {
-		line := scanner.Text()
-		if strings.Contains(line, match) {
-			return true
-		}
+func existsKernel() bool {
+	// from `MSI` or `Windows Update`
+	kernel := filepath.Join(getSystem32Root(), "lxss", "tools", "kernel")
+	if err := fileutils.Exists(kernel); err == nil {
+		return true
+	}
+
+	// from `Microsoft Store` or `Github`
+	kernel = filepath.Join(getProgramFiles(), "WSL", "tools", "kernel")
+	if err := fileutils.Exists(kernel); err == nil {
+		return true
 	}
+
 	return false
 }
+
+func getLocalAppData() (result string, ok bool) {
+	if local := os.Getenv("LOCALAPPDATA"); local != "" {
+		return local, true
+	}
+
+	if user := os.Getenv("USERPROFILE"); user != "" {
+		return filepath.Join(user, "AppData", "Local"), true
```

Comment:
is this defined anywhere as a constant; or is it something we use more than once around our code?

Reply:
> is this defined anywhere as a constant;
Not defined elsewhere

> is it something we use more than once around our code?
Currently, it is only being used here.